### PR TITLE
Michael user registration validation

### PR DESCRIPTION
**Pull request by MaybeFreak**

### CODE REVIEW

File: src/pages/dashboard/index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const Dashboard = () => {
 					</div>
 				</Card>
 
-				<Posts />
+				{/* <Posts /> */}
```
Comment (vherus) [inline src/pages/dashboard/index.js]: Let's uncomment this. A fix has been merged into the server so we should be good now

File: src/pages/register/index.js [hunk 2]
```diff
@@ -8,12 +8,42 @@ import "./register.css";
 const Register = () => {
 	const { onRegister } = useAuth();
 	const [formData, setFormData] = useState({ email: "", password: "" });
+	const [isEmailValid, setIsEmailValid] = useState(null)
+	const [isPasswordValid, setIsPasswordValid] = useState(null)
+	const [isEmailTaken, setIsEmailTaken] = useState(null)
```
Comment (vherus) [inline src/pages/register/index.js:14]: Boolean state should be initialised as either true or false, I'd change these nulls to false

File: src/pages/register/index.js [hunk 3]
```diff
@@ -8,12 +8,42 @@ import "./register.css";
...
+			if (email.length > 254) return false
+			
+			const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+			return regex.test(email)
+		}
+
+		const validatePassword = (password) => {
+			const regex = /^(?=.*[\d])(?=.*[!@#$%^&*])[\w!@#$%^&*]{8,}$/
+			return regex.test(password)
+		}
```
Comment (vherus) [inline src/pages/register/index.js]: These functions will be useful for the login for as well, I'd move these to a different file where any other component can use them. Perhaps `utils` folder with a `validation.js` file inside?

**Review by vherus — CHANGES_REQUESTED** [review]
Nice work. There are some changes to make before this gets merged

File: src/pages/register/index.js (hunk 2 above)
Comment (MaybeFreak) [reply src/pages/register/index.js:14]: Would have to set them to true by default then so the error messages don't show up on first load only after submit has complaints.

File: src/pages/register/index.js (hunk 3 above)
Comment (MaybeFreak) [reply src/pages/register/index.js]: Should I then create a new folder in `src` called `utils`, don't see one already set up.  Or should i add the `validation.js`into the `./src/service` folder ?

File: src/pages/register/index.js (hunk 2 above)
Comment (vherus) [reply src/pages/register/index.js:14]: Ah, good point. In that case you can ignore my comment on this

File: src/pages/register/index.js (hunk 3 above)
Comment (vherus) [reply src/pages/register/index.js]: Yeah, you can create a utils folder for this

**Review by vherus — APPROVED** [review]

**Review by WestyCodes — APPROVED** [review]

**Pull request merged**